### DISCUSSION TITLE
A few basic questions please - about the done method and deleting a queue from the Redis

### DISCUSSION THREAD

**Discussion by Taxi4you** [OptimalBits/bull#2406] [General]
Its not a bug but a few questions please:

1. Should I call the method "done()" in order to consider a job has been completed, or it will be considered done by the time that there no operations to perform in the process callback?
2. How to delete a queue or update a cronjob setting for a current queue? For example: I have a queue configured to run by a cronjob for every minute and I want to update it for just the minute 10. How this can be done? Should I go to redis and flushall everytime I need such a modification?

**Comment by manast** [comment]
> 1. Should I call the method "done()" in order to consider a job has been completed, or it will be considered done by the time that there no operations to perform in the process callback?

You can either use done callback or promises: https://docs.bullmq.io/bull/quick-guide#basic-usage




> 2\. How to delete a queue or update a cronjob setting for a current queue? For example: I have a queue configured to run by a cronjob for every minute and I want to update it for just the minute 10. How this can be done? Should I go to redis and flushall everytime I need such a modification?

You should be able to find answers to these questions here: https://github.com/OptimalBits/bull/blob/develop/REFERENCE.md

**Reply by Taxi4you** [reply]
But if I don't run the "done" method and not returning a promise - just void. Will it still work properly?

**Reply by manast** [reply]
If you define a "done" callback you need to use it, otherwise you must return a promise.

**Reply by Taxi4you** [reply]
I understand, my question is if I don't define a "done" callback and also not returning anything, just void function without any params.

**Reply by manast** [reply]
But I already answered, you either provide a "done" callback and use it or you return a promise.
